apps/api/src/services/competition.service.ts

```diff
@@ -1592,8 +1600,89 @@ export class CompetitionService {
       throw new Error(`Competition not found: ${competitionId}`);
     }
 
+    // If perpsProvider is provided but type is not, auto-set type to perpetual_futures
+    if (perpsProvider && !updates.type) {
+      updates.type = "perpetual_futures";
+    }
+
+    // Check if type is being changed
+    const isTypeChanging =
+      updates.type !== undefined && updates.type !== existingCompetition.type;
+
+    if (isTypeChanging) {
+      // Only allow type changes for pending competitions
+      if (existingCompetition.status !== "pending") {
```

Comment:
PR description suggests that changing type should be forbidden if voting is open, but I don't see a check for that here?  I believe voting can be open when a competition is still in pending state

Reply:
imo I _dint think_ it'd be a big deal if voting was open, and the comp type was changed. Mainly, because this should be a rare occurrence and only occur if we screw something up.

Reply:
Sorry - was a mistake in the description. We would want the admin to be able to change the type even if voting is still open (which, I believe, applies to our first use case of this particular endpoint for this upcoming competition)